tests/integration/golang/aim/app/delete_app_test.go

```diff
@@ -0,0 +1,114 @@
+//go:build integration
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type DeleteAppTestSuite struct {
+	suite.Suite
+	client      *helpers.HttpClient
+	appFixtures *fixtures.AppFixtures
+	apps        []*models.App
+}
+
+func TestDeleteAppTestSuite(t *testing.T) {
+	suite.Run(t, new(DeleteAppTestSuite))
+}
+
+func (s *DeleteAppTestSuite) SetupTest() {
+	s.client = helpers.NewAimApiClient(os.Getenv("SERVICE_BASE_URL"))
+
+	appFixtures, err := fixtures.NewAppFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.appFixtures = appFixtures
+
+	s.apps, err = s.appFixtures.CreateTestApps(context.Background(), 10)
+	assert.Nil(s.T(), err)
+}
+
+func (s *DeleteAppTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.appFixtures.UnloadFixtures())
+	}()
+	tests := []struct {
+		name        string
+		wantAppCount int
+	}{
+		{
+			name: "DeleteApp",
+			wantAppCount: 9,
+		},
+	}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+
+			var deleteResponse map[string]any
+			err := s.client.DoDeleteRequest(
+				fmt.Sprintf("/apps/%s", s.apps[0].ID),
+				&deleteResponse,
+			)
+			assert.Nil(s.T(), err)
+
+			var getResponse []models.App
+			err = s.client.DoGetRequest(
```

Comment:
in each `integration` test we try to be atomic and test only one particular endpoint. other calls has to go directly do database. can you please change it?

Reply:
the big problem of this approach which I faced before(in some previous projects), that if endpoint signature changed(endpoint which is in use in many places) you probably have to adjust many many tests(all the tests which use this endpoint). When you use database instead, of course something could also be changed but much more rarely and  usually you have to change the query which is under the hood of `methods`. In that case you probably have to do almost nothing to catch new changes. So that's why to have atomic test which tests exactly one `endpoint` instead of to have a test, which call other 2-3-4 endpoints to test actual endpoint.